test/e2e_node/memory_manager_test.go

```diff
@@ -319,6 +333,35 @@ var _ = SIGDescribe("Memory Manager [Serial] [Feature:MemoryManager][NodeAlphaFe
 				},
 			}
 		})
+		ginkgo.When("pod has init containers", func() {
+			ginkgo.BeforeEach(func() {
+				//override init container parameters
+				initCtnParams = []memoryManagerCtnAttributes{
+					{
+						ctnName: "init-memory-manager-static",
+						cpus:    "100m",
+						memory:  "128Mi",
+					},
+				}
+			})
```

Comment:
It looks alright... The only thing I wonder is if we should create such a nesting test. Maybe we can add ginkgo.When() also for the second test case? To have similar nesting? Setting `ctnParams` twice may not be a bad solution if it improves readability. 

Something like:
```
ginkgo.Context("with static policy", func() {
	ginkgo.BeforeEach(func() {
		// override kubelet configuration parameters
		tmpParams := *defaultKubeParams
		tmpParams.memoryManagerPolicy = staticPolicy
		kubeParams = &tmpParams
	})

	ginkgo.JustAfterEach(func() {
		// reset containers attributes
		ctnParams = []memoryManagerCtnAttributes{}
		initCtnParams = []memoryManagerCtnAttributes{}
	})

	ginkgo.When("pod has init and app containers", func() {
		ginkgo.BeforeEach(func() {
			// override containers parameters
			ctnParams = []memoryManagerCtnAttributes{
				{
					ctnName: "memory-manager-static",
					cpus:    "100m",
					memory:  "128Mi",
				},
			}
			// override init container parameters
			initCtnParams = []memoryManagerCtnAttributes{
				{
					ctnName: "init-memory-manager-static",
					cpus:    "100m",
					memory:  "128Mi",
				},
			}
		})

		ginkgo.It("should succeed to start the pod", func() {
			ginkgo.By("Running the test pod")
			testPod = f.PodClient().CreateSync(testPod)

			// it no taste to verify NUMA pinning when the node has only one NUMA node
			if !*isMultiNUMASupported {
				return
			}

			verifyMemoryPinning([]int{0})
		})
	})

	ginkgo.When("pod has only app containers", func() {

		ginkgo.BeforeEach(func() {
			// override containers parameters
			ctnParams = []memoryManagerCtnAttributes{
				{
					ctnName: "memory-manager-static",
					cpus:    "100m",
					memory:  "128Mi",
				},
			}
		})
		
		ginkgo.It("should succeed to start the pod", func() {
			ginkgo.By("Running the test pod")
			testPod = f.PodClient().CreateSync(testPod)

			// it no taste to verify NUMA pinning when the node has only one NUMA node
			if !*isMultiNUMASupported {
				return
			}

			verifyMemoryPinning([]int{0})
		})
	})
})
```
The next thing is that the common part of two test cases can be extracted to some function that takes container params.

Reply:
works for me